lib/logstash-docket/plugin.rb

```diff
@@ -78,6 +78,18 @@ def desc
       fail NotImplementedError
     end
 
+    def with_alias(alias_mappings)
+      yield self
+
+      if alias_mappings.include?(type) && alias_mappings[type].include?(name)
+        alias_mappings[type][name].each do |alias_name|
+          yield AliasPlugin.new(canonical_plugin: self,
+                                alias_name: alias_name["alias"],
+                                doc_headers: alias_name["doc_headers"])
+        end
+      end
```

Comment:
`FLAT_ACCESS` alternative:

```suggestion
      alias_mappings.fetch("#{type}-#{name}", []).each do |alias_definition|
        yield AliasPlugin.new(canonical_plugin: self,
                                alias_name: alias_definition["alias"],
                                doc_headers: alias_definition["doc_headers"])
        end
      end
```

Reply:
It seems I will need a time to get familiar with `.dig` :)) So, for the safety used return logic like this: `return unless alias_definitions.key?(type)`. Let me please know if this can be still simplified.